### PR TITLE
MultiDB support

### PR DESCRIPTION
**Pull request by realityking**
As requested by Mark here's pull with all the changes needed for the CMS's MultiDB work.

### CODE REVIEW

**Comment by dextercowley** [comment]
I hope this can be committed soon. We would very much like to get this in before the CMS 2.5 beta (so platform 11.4?). Thanks!

**Comment by eddieajau** [comment]
I haven't finished reviewing the work from Sudhi, I was just getting it into a condition where it was reviewable.  Don't merge yet - it's not ready.

**Comment by eddieajau** [comment]
Ok, can you pull from my sudhi-staging branch.  It should be all up to date.

File: libraries/joomla/database/table/user.php [hunk 1]
```diff
@@ -465,7 +465,7 @@ public function setLastVisit($timeStamp = null, $userId = null)
 		$db = $this->_db;
 		$query = $db->getQuery(true);
 		$query->update($db->quoteName($this->_tbl));
-		$query->set($db->quoteName('lastvisitDate') . '=' . $db->quote($date->format($db->getDateFormat())));
+		$query->set($db->quoteName('lastvisitDate') . '=' . $db->quote($date->format($this->_db->getDateFormat())));
 		$query->where($db->quoteName('id') . '=' . (int) $userId);
```
Comment (chdemko) [inline libraries/joomla/database/table/user.php]:
```
$date->format($this->_db->getDateFormat())
```

could be replaced by

```
$date->toSql()
```

Comment (eddieajau) [inline libraries/joomla/database/table/user.php]: Fixed in my branch.

**Comment by realityking** [comment]
I pulled in the work from your branch and hopefully fixed all the the code style errors.

**Comment by eddieajau** [comment]
Errors

libraries/joomla/database/databasequery.php:810
The comments for parameters $table (1) and $increment_field (2) do not align

libraries/joomla/database/databasequery.php:810
Missing comment for param "$increment_field" at position 2

libraries/joomla/database/table/usergroup.php:217
Expected "foreach (...)\n...{\n"; found "foreach(...)\n...{\n"

libraries/joomla/database/table/usergroup.php:231
Concat operator must be followed by one space

libraries/joomla/database/table/usergroup.php:231
Concat operator must be preceeded by one space

libraries/joomla/database/table/usergroup.php:231
Concat operator must be followed by one space

libraries/joomla/database/table/usergroup.php:231
Concat operator must be preceeded by one space

**Comment by eddieajau** [comment]
I give up.  The hooduku commits put back a heap of stuff that was unnecessary or wrong.  Please revert them.

**Comment by dextercowley** [comment]
Hi Andrew. I didn't commit anything to the platform, so I'm not sure
what you are asking me to do. I'm on line if you want to chat. Thanks.
Mark

On Thu, Dec 15, 2011 at 2:03 PM, Andrew Eddie
reply@reply.github.com
wrote:

**Comment by eddieajau** [comment]
88d7b8a and 18c9d57 need to be taken out of this branch.  Nothing you did or can do.

**Comment by elinw** [comment]
I suggest closing this pull request. If  Andrew is really giving up then we can make a new one just based on Andrew's plus the code style clean up I did in my pull of Andrew's  branch, assuming that nothing works worse with that than what we have now. I think the messiness of  this is more about people not understanding how to work with git than any ill-intent.

**Comment by eddieajau** [comment]
Um, let me be clear. "I give up" is meant in terms of exasperation.  I have spent significant spare time on this already but don't have any bandwidth left to sort out whatever has happened this close to Christmas.  Sorry if that doesn't suit but that's the way it is.

**Comment by elinw** [comment]
Yeah we all understand that feeling.

**Pull request closed without merging**